### ISSUE TITLE
Training error with Yolov3 5 yolo output layers

### ISSUE THREAD

**Issue by tpereztorres** [AlexeyAB/darknet#2202]
Hello @AlexeyAB,

We are trying to use Yolov3 with 5 Yolo output layers (yolov3_5l.cfg) but it does not train well because  the loss in the second step starts to be infinity as it can be seen in the next image.

![image](https://user-images.githubusercontent.com/22599487/51108886-c7e9f080-17f3-11e9-8e2f-0d230891ff5d.png)

We have tested with other anchors, batch size, subdivisions and learning rate but the error still persists.

Could you help us in order to be able to train this network?

Thank you very much.

**Comment by AlexeyAB** [comment]
@tpereztorres Hello,

I **can't** reproduce your issue.

I use:
* Windows 7 x64
* CUDA 10.0
* cuDNN 7.1.3 for CUDA 10.0
* OpenCV 3.3.0
* MSVS 2015
* GPU RTX 2070
* compiled without CUDNN_HALF


I use this command:
`darknet.exe detector train cfg/coco_my.data cfg/yolov3_5l.cfg darknet53.conv.74 -map`

I use this default yolov3_5l.cfg file, just set `subdivisions=32`: https://github.com/AlexeyAB/darknet/blob/master/build/darknet/x64/cfg/yolov3_5l.cfg

I can train at least 300 iterations and get avg_loss = 30:
![image](https://user-images.githubusercontent.com/4096485/51113836-861d7200-1814-11e9-83f9-e6e3b2e88485.png)

**Comment by AlexeyAB** [comment]
After 1000 iterations, avg_loss ~= 4.2

![image](https://user-images.githubusercontent.com/4096485/51126022-1965a000-1833-11e9-8283-0ad5e9ed09cc.png)

**Comment by AlexeyAB** [comment]
I can get `Nan`, only if I comment line `burn_in=1000` in cfg-file.

So just **don't disable `burn_in=1000`** in cfg-file.

![image](https://user-images.githubusercontent.com/4096485/51126413-1f0fb580-1834-11e9-821c-f469d3c0e8fc.png)

**Comment by tpereztorres** [comment]
Hello again @AlexeyAB,

First, thank you for all the information.

We use:

- Windows 10 x64

- CUDA 9.1

- cuDNN 7.1 for CUDA 9.1

- compiled with CUDNN_HALF

- commit from October.

Could the error come from one of these characteristics?

We have not disable **burn_in** parameter in cfg-file.

Thank you very much.

**Comment by AlexeyAB** [comment]
@tpereztorres 

* What GPU do you use?

* Just try to use the latest version of Darknet

* If you don't want to use new Darknet version, then:
    * or comment this line: https://github.com/AlexeyAB/darknet/blob/08e8e0c8c2d3cddfb6ee065e813ae5023f9568d2/src/parser.c#L658
    * or compile without CUDNN_HALF

**Comment by tpereztorres** [comment]
Hello again @AlexeyAB,

Thanks for the quick answer.

We use a GPU RTX 2080 Ti.

We will try it with the latest version of Darknet, commenting the line that you specify or compiling without CUDNN_HALF.

Just one question, will the compilation without CUDNN_HALF affect the results of the prediction?

Thank you very much.

**Comment by AlexeyAB** [comment]
> Just one question, will the compilation without CUDNN_HALF affect the results of the prediction?

* For training
    * In the old versions, CUDNN_HALF gives lower final accuracy during training, than without CUDNN_HALF.
    * In the new version with CUDNN_HALF and without gives the same good accuracy during training.

* For detection with CUDNN_HALF and without gives the same accuracy.

With CUDNN_HALF the Darknet is faster on GPU RTX.

**Comment by tpereztorres** [comment]
Hello @AlexeyAB,

Perfect, thank you!
